ufs2arco/regrid/mom6regridder.py

```diff
@@ -62,16 +63,58 @@ def _create_regridder(self, ds_in: xr.Dataset) -> None:
                 f"Vector fields will be silently ignored."
             )
 
+        # create input dataset
+        mom6_grid = self.config.get("rotation_file", None)
+        mom6_grid = xr.open_dataset(mom6_grid)
+        grid_in = xr.Dataset()
+        grid_in["lon"] = mom6_grid["geolon"]
+        grid_in["lat"] = mom6_grid["geolat"]
+        grid_in["lon_u"] = mom6_grid["geolon_u"]
+        grid_in["lat_u"] = mom6_grid["geolat_u"]
+        grid_in["lon_v"] = mom6_grid["geolon_v"]
+        grid_in["lat_v"] = mom6_grid["geolat_v"]
+        ny, nx = grid_in["lon"].shape
+        lon_b = np.empty((ny + 1, nx + 1))
+        lat_b = np.empty((ny + 1, nx + 1))
+        lon_b[1:, 1:] = mom6_grid["geolon_c"].values
+        lat_b[1:, 1:] = mom6_grid["geolat_c"].values
+        # periodicity
+        lon_b[:, 0] = lon_b[:, -1]
+        lat_b[:, 0] = lat_b[:, -1]
+        # south edge
+        dy = (lat_b[2, :] - lat_b[1, :]).mean()
+        lat_b[0, 1:] = lat_b[1, 1:] - dy
+        lon_b[0, 1:] = lon_b[1, 1:]
+        # corner point
+        lon_b[0, 0] = lon_b[1, 0]
+        lat_b[0, 0] = lat_b[0, 1]
+        grid_in["lon_b"] = xr.DataArray(data=lon_b)
+        grid_in["lat_b"] = xr.DataArray(data=lat_b)

```

Comment:
Soft suggestion: we could put this in a helper method/function to declutter. Up to you though

Reply:
yeah good call. was thinking this myself but just didn't get to it. can definitely do that